### PR TITLE
[Ready to merge] Pruned_transducer_stateless2 for alimeeting dataset

### PR DESCRIPTION
**Pull request by luomingshuang**
This PR has been tested with one gpu. I will find some free gpus to finish it.

### CODE REVIEW

File: egs/alimeeting/ASR/local/compute_fbank_alimeeting.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+            cur_num_jobs = num_jobs if ex is None else 80
+            cur_num_jobs = min(cur_num_jobs, len(cut_set))
+
+            cut_set = cut_set.compute_and_store_features(
+                extractor=extractor,
+                storage_path=f"{output_dir}/feats_{partition}",
+                # when an executor is specified, make more partitions
+                num_jobs=cur_num_jobs,
+                executor=ex,
+                storage_type=LilcomHdf5Writer,
```
Comment (csukuangfj) [inline egs/alimeeting/ASR/local/compute_fbank_alimeeting.py:89]: This may explain why your GPU usage is very low.

Please use
https://github.com/k2-fsa/icefall/blob/c4ee2bc0af8a758a8ae049f72619a111fb0667e2/egs/librispeech/ASR/local/compute_fbank_librispeech.py#L88

Comment (csukuangfj) [reply egs/alimeeting/ASR/local/compute_fbank_alimeeting.py:89]: Please also change `aidatatang_200zh`. It has the same issue.

**Comment by luomingshuang** [comment]
In this PR, I trained pruned rnnt2 with alimeeting far(the default mode) data. There are some results:
The CERs(%) are
|                                    |     eval   |    test    | comment                                  |
|------------------------------------|------------|------------|------------------------------------------|
|          greedy search             | 31.77      | 34.66      | --epoch 29, --avg 18, --max-duration 100 |
| modified beam search (beam size 4) | 30.38      | 33.02      | --epoch 29, --avg 18, --max-duration 100 |
| fast beam search (set as default)  | 31.39      | 34.25      | --epoch 29, --avg 18, --max-duration 1500|

The eval baseline result from the [paper](https://arxiv.org/pdf/2110.07393.pdf) trained with the alimeeting far data is 34.3% (CER).

File: egs/alimeeting/ASR/local/compute_fbank_alimeeting.py (hunk 1 above)
Comment (luomingshuang) [reply egs/alimeeting/ASR/local/compute_fbank_alimeeting.py:89]: Oh, thanks.

**Comment by csukuangfj** [comment]
Results from that paper
Screen Shot 2022-06-01 at 14 16 37

**Comment by csukuangfj** [comment]
Could you try the solution proposed at https://github.com/docker-library/python/issues/731#issue-1249165685

That is, change
https://github.com/k2-fsa/icefall/blob/fbfc98f1d37b0f4c8b58307eff3b5f82808e88fb/.github/workflows/test.yml#L79
to
```bash
pip install -r requirements.txt 

pip uninstall protobuf
pip install --no-binary protobuf protobuf
```

If it does not work, you can try

```bash
pip uninstall protobuf
pip install --no-binary protobuf protobuf

pip install -r requirements.txt 
```

**Comment by luomingshuang** [comment]
Done. https://huggingface.co/luomingshuang/icefall_asr_alimeeting_pruned_transducer_stateless2/blob/main/exp/cpu_jit_torch.1.11.0.pt

**Comment by csukuangfj** [comment]
Could you use torch 1.10.0 as I am using torch 1.10.0 on huggingface.

**Comment by luomingshuang** [comment]
is 1.7.0 OK?

**Comment by csukuangfj** [comment]
> is 1.7.0 OK?

Yes, that should be fine.

**Comment by csukuangfj** [comment]
> https://huggingface.co/luomingshuang/icefall_asr_alimeeting_pruned_transducer_stateless2/blob/main/exp/cpu_jit_torch_1.7.1.pt

Thanks! I just added  the pre-trained model of alimeeting to https://huggingface.co/spaces/k2-fsa/automatic-speech-recognition

**Comment by kjw11** [comment]
> In this PR, I trained pruned rnnt2 with alimeeting far(the default mode) data. There are some results: The CERs(%) are
> 
> eval	test	comment
> greedy search	31.77	34.66	--epoch 29, --avg 18, --max-duration 100
> modified beam search (beam size 4)	30.38	33.02	--epoch 29, --avg 18, --max-duration 100
> fast beam search (set as default)	31.39	34.25	--epoch 29, --avg 18, --max-duration 1500
> The eval baseline result from the [paper](https://arxiv.org/pdf/2110.07393.pdf) trained with the alimeeting far data is 34.3% (CER).

@luomingshuang Thanks for your contribution! I notice the results of this code are compared to the SOT results of M2MET papers.

Just want to understand the comparison, do you regard the code of this PR as a naive ASR task, or like [M2MET challenge](https://arxiv.org/pdf/2202.03647.pdf), as a multi-speaker ASR task? Since the latter task sort of additionally requires the speech should be transcripted with speaker identities ([SOT](https://arxiv.org/pdf/2003.12687.pdf) style), the CER contains a stricter constrain. Alternatively, if this is a single-speaker ASR setting,  the CER is not fully comparable to M2MET case.

Thanks again for your contribution.

**Comment by luomingshuang** [comment]
Oh，this is just a naive ASR task, not a multi-speaker ASR task. I use the `train-Ali-far` as my training data and don't consider the speaker id. So you are true. The 34.3% of CER is a multi-speaker ASR's result of TABLE 5 in [paper](https://arxiv.org/pdf/2110.07393.pdf). Thanks for pointing out this.

**Comment by desh2608** [comment]
BTW the original M2MeT challenge also used some text normalization that is not currently present in the Lhotse AliMeeting recipe. I have created a PR to add it [here](https://github.com/lhotse-speech/lhotse/pull/909).
